solidity-frontend/src/main/java/org/sonarsource/solidity/frontend/SolidityTokensInfo.java

```diff
@@ -0,0 +1,81 @@
+    "assemblyBlock", "assemblyItem", "assemblyExpression", "assemblyCall",
+    "assemblyLocalDefinition", "assemblyAssignment", "assemblyIdentifierOrList",
+    "assemblyIdentifierList", "assemblyStackAssignment", "labelDefinition",
+    "assemblySwitch", "assemblyCase", "assemblyFunctionDefinition", "assemblyFunctionReturns",
+    "assemblyFor", "assemblyIf", "assemblyLiteral", "subAssembly", "tupleExpression",
+    "elementaryTypeNameExpression", "numberLiteral", "identifier"
+  };
+
+  private static final String[] _LITERAL_NAMES = {
+    null, "'pragma'", "';'", "'^'", "'~'", "'>='", "'>'", "'<'", "'<='", "'='",
+    "'as'", "'import'", "'*'", "'from'", "'{'", "','", "'}'", "'contract'",
+    "'interface'", "'library'", "'is'", "'('", "')'", "'using'", "'for'",
+    "'struct'", "'constructor'", "'modifier'", "'function'", "'returns'",
+    "'event'", "'enum'", "'['", "']'", "'.'", "'mapping'", "'=>'", "'memory'",
+    "'storage'", "'if'", "'else'", "'while'", "'assembly'", "'do'", "'return'",
+    "'throw'", "'emit'", "'var'", "'address'", "'bool'", "'string'", "'byte'",
+    "'++'", "'--'", "'new'", "'+'", "'-'", "'after'", "'delete'", "'!'", "'**'",
+    "'/'", "'%'", "'<<'", "'>>'", "'&'", "'|'", "'=='", "'!='", "'&&'", "'||'",
+    "'?'", "':'", "'|='", "'^='", "'&='", "'<<='", "'>>='", "'+='", "'-='",
+    "'*='", "'/='", "'%='", "'let'", "':='", "'=:'", "'switch'", "'case'",
+    "'default'", "'->'", null, null, null, null, null, null, null, null, null,
+    null, null, null, "'anonymous'", "'break'", "'constant'", "'continue'",
+    "'external'", "'indexed'", "'internal'", "'payable'", "'private'", "'public'",
+    "'pure'", "'view'"
+  };
+  private static final String[] _SYMBOLIC_NAMES = {
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, null, null, null, null, null, null,
+    null, null, null, null, null, null, "Int", "Uint", "Byte", "Fixed", "Ufixed",
+    "VersionLiteral", "BooleanLiteral", "DecimalNumber", "HexNumber", "NumberUnit",
+    "HexLiteral", "ReservedKeyword", "AnonymousKeyword", "BreakKeyword", "ConstantKeyword",
+    "ContinueKeyword", "ExternalKeyword", "IndexedKeyword", "InternalKeyword",
+    "PayableKeyword", "PrivateKeyword", "PublicKeyword", "PureKeyword", "ViewKeyword",
+    "Identifier", "StringLiteral", "WS", "COMMENT", "LINE_COMMENT"
+  };
+  public static final Vocabulary VOCABULARY = new VocabularyImpl(_LITERAL_NAMES, _SYMBOLIC_NAMES);
+
+  @Deprecated
+  public String[] getTokenNames() {
+    return null;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Return an empty array instead of null. ([squid:S1168](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=13&issues=AWP4T8oV23Tx-xDbYyiZ&open=AWP4T8oV23Tx-xDbYyiZ)

[ISSUE_KEY]: AWP4T8oV23Tx-xDbYyiZ